twister: add option to ignore skipped tests in reports

nashif opened the pull request:
With --no-skipped, twister will skip over all tests marked as skipped
(filtered) in the junit report.

This is useful for CI where have 1000s of filtered tests that appear in
the report and in some cases cause tools parsing the output to fail or
provide incomplete results.

Fixes #38179

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: add option to ignore skipped tests in reports

With --no-skipped-report, twister will skip over all tests marked as
skipped (filtered) in the junit report.

This is useful for CI where have 1000s of filtered tests that appear in
the report and in some cases cause tools parsing the output to fail or
provide incomplete results.

Fixes #38179

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

ceolin commented on scripts/pylib/twister/twisterlib.py:
```diff
@@ -3653,17 +3653,20 @@ def xunit_report(self, filename, platform=None, full_report=False, append=False,
-                # Multiple 'property' can be added to 'properties'
-                # differing by name and value
-                ET.SubElement(eleTSPropetries, 'property', name="version", value=version)
+                if not no_skipped or total != skips:
```
`or total != skips:` will not cause it showing some skipped tests ?

cfriedt replied:
might be able to eliminate the double-negative if you rename the variable to `report_skipped`.

cfriedt reviewed (dismissed):
Needs a rebase, but LGTM

PerMac reviewed (changes requested):
Could you please add the same option to json and csv reports for consistency? Sorry for so late review, I missed this PR somehow.

PerMac reviewed (dismissed):
I changed to approve as I don't wont to block this PR. However, the consistency in the reports is important, so please add such option also for json and csv

enjiamai reviewed (dismissed):
I think we will need this feature, and this only enabled when --no-skipped used.

mbolivar-nordic commented:
@nashif please rebase

nashif committed:
ci: twister: do not report skipped tests

Do not report skipped tests to reduce size of report and to make it
parseable.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
